# H001 — الانعكاس قصير المدى (Short-Term Reversal)

هذا الدفتر فرضية واحدة محدَّدة — يعتمد على دفتر المحور الأساسي `signal_evaluation_axis (3).ipynb` (IC، فحص عُشر، خطّ أساس عشوائي، `evaluate_windows`، وسجلّ التجارب `register_hypothesis`/`list_registry`) بدل تكراره. شغّل الخلية التالية أولاً (أو تأكد من تشغيل الدفتر الأساسي في نفس الجلسة) قبل أي خلية أخرى هنا.

أُفرِد هذا الملف من `signal_evaluation_axis (3).ipynb` الأصلي (كان يضمّ المحور والفرضيات معاً في ملف واحد كبير) — نفس المحتوى حرفياً، بلا أي تعديل على الأرقام أو المنطق.

In [ ]:
%run "signal_evaluation_axis (3).ipynb"

## ١٠) تسجيل فرضية الانعكاس قصير المدى — أول سجلّ فعلي

نتيجة تشغيل بوّابة المرحلة ٠ الفعلي على بياناتك (٣٠ نافذة متحرّكة،
`momentum_predict_fn`، `target_key='y_close_reg'`، أفق يوم واحد): الإشارة
معكوسة الاتجاه بثبات — 22 من 30 نافذة سالبة، كثير منها بدلالة إحصائية شبه
مطلقة. هذا يعني أمرين معاً: (أ) المحور موثوق — التقط إشارة حقيقية قوية على
فرضية معروفة الاتجاه العام أكاديمياً، بوّابة المرحلة ٠ اجتازت. (ب) الفرضية
الفعلية المؤكَّدة ليست "زخم" بل **انعكاس قصير المدى** — مُدرَجة أصلاً في
جدول فرضيات الأساس بالمرحلة ١، والآن تملك دليلاً رقمياً لا افتراضاً.

يُسجَّل التالي **بالأرقام المُبلَّغة فعلياً** من التشغيل — لا يُعاد حسابها هنا
(هذا الدفتر لا يملك بياناتك الحقيقية).

In [ ]:
# @title
# نتائج التشغيل الفعلي المُبلَّغة (٣٠/٣٠ نافذة ناجحة) — منسوخة كما وردت، لا مُعاد حسابها.
_per_window_actual = [
    {"window": "نافذة 1",  "status": "ok", "ic": -0.290746, "spread": -0.028182, "monotonic": False, "p_value": 0.000, "percentile": 0.0},
    {"window": "نافذة 2",  "status": "ok", "ic":  0.026592, "spread": -0.003820, "monotonic": False, "p_value": 0.425, "percentile": 79.1},
    {"window": "نافذة 3",  "status": "ok", "ic": -0.084153, "spread": -0.005970, "monotonic": False, "p_value": 0.007, "percentile": 0.4},
    {"window": "نافذة 4",  "status": "ok", "ic": -0.102203, "spread": -0.015976, "monotonic": False, "p_value": 0.000, "percentile": 0.0},
    {"window": "نافذة 5",  "status": "ok", "ic": -0.020439, "spread": -0.007911, "monotonic": False, "p_value": 0.422, "percentile": 21.5},
    {"window": "نافذة 6",  "status": "ok", "ic": -0.081024, "spread": -0.021007, "monotonic": False, "p_value": 0.000, "percentile": 0.0},
    {"window": "نافذة 7",  "status": "ok", "ic": -0.135428, "spread": -0.016676, "monotonic": False, "p_value": 0.000, "percentile": 0.0},
    {"window": "نافذة 8",  "status": "ok", "ic": -0.068493, "spread": -0.018452, "monotonic": False, "p_value": 0.005, "percentile": 0.2},
    {"window": "نافذة 9",  "status": "ok", "ic": -0.127898, "spread": -0.014369, "monotonic": False, "p_value": 0.000, "percentile": 0.0},
    {"window": "نافذة 10", "status": "ok", "ic": -0.180442, "spread": -0.159693, "monotonic": False, "p_value": 0.000, "percentile": 0.0},
    {"window": "نافذة 11", "status": "ok", "ic": -0.003948, "spread":  0.001643, "monotonic": False, "p_value": 0.857, "percentile": 44.9},
    {"window": "نافذة 12", "status": "ok", "ic": -0.087822, "spread": -0.015369, "monotonic": False, "p_value": 0.000, "percentile": 0.0},
    {"window": "نافذة 13", "status": "ok", "ic": -0.145143, "spread": -0.035949, "monotonic": False, "p_value": 0.000, "percentile": 0.0},
    {"window": "نافذة 14", "status": "ok", "ic":  0.079888, "spread":  0.024450, "monotonic": False, "p_value": 0.000, "percentile": 100.0},
    {"window": "نافذة 15", "status": "ok", "ic": -0.115865, "spread": -0.006311, "monotonic": False, "p_value": 0.000, "percentile": 0.0},
    {"window": "نافذة 16", "status": "ok", "ic": -0.080590, "spread": -0.002425, "monotonic": False, "p_value": 0.000, "percentile": 0.0},
    {"window": "نافذة 17", "status": "ok", "ic":  0.018620, "spread":  0.015342, "monotonic": False, "p_value": 0.376, "percentile": 81.9},
    {"window": "نافذة 18", "status": "ok", "ic":  0.086994, "spread":  0.021201, "monotonic": False, "p_value": 0.000, "percentile": 100.0},
    {"window": "نافذة 19", "status": "ok", "ic": -0.087024, "spread": -0.004437, "monotonic": False, "p_value": 0.000, "percentile": 0.0},
    {"window": "نافذة 20", "status": "ok", "ic": -0.150789, "spread": -0.025203, "monotonic": False, "p_value": 0.000, "percentile": 0.0},
    {"window": "نافذة 21", "status": "ok", "ic": -0.189262, "spread": -0.048517, "monotonic": False, "p_value": 0.000, "percentile": 0.0},
    {"window": "نافذة 22", "status": "ok", "ic": -0.152104, "spread": -0.001856, "monotonic": False, "p_value": 0.000, "percentile": 0.0},
    {"window": "نافذة 23", "status": "ok", "ic": -0.050583, "spread": -0.006933, "monotonic": False, "p_value": 0.004, "percentile": 0.4},
    {"window": "نافذة 24", "status": "ok", "ic": -0.071903, "spread":  0.011298, "monotonic": False, "p_value": 0.000, "percentile": 0.0},
    {"window": "نافذة 25", "status": "ok", "ic": -0.043449, "spread":  0.007428, "monotonic": False, "p_value": 0.020, "percentile": 1.0},
    {"window": "نافذة 26", "status": "ok", "ic": -0.215032, "spread": -0.026710, "monotonic": False, "p_value": 0.000, "percentile": 0.0},
    {"window": "نافذة 27", "status": "ok", "ic":  0.025248, "spread":  0.008714, "monotonic": False, "p_value": 0.160, "percentile": 92.3},
    {"window": "نافذة 28", "status": "ok", "ic": -0.087753, "spread": -0.001497, "monotonic": False, "p_value": 0.000, "percentile": 0.0},
    {"window": "نافذة 29", "status": "ok", "ic": -0.073893, "spread": -0.007213, "monotonic": False, "p_value": 0.000, "percentile": 0.0},
    {"window": "نافذة 30", "status": "ok", "ic": -0.007333, "spread": -0.000805, "monotonic": False, "p_value": 0.667, "percentile": 34.0},
]

_actual_report = {
    "mean_ic": -0.0805, "std_ic": 0.0835, "frac_significant": 0.73,
    "consistent_sign": False, "n_ok": 30,
    "per_window": _per_window_actual,
}


class _PrebuiltReport(dict):
    """يحاكي واجهة report['per_window'].to_dict — القيمة هنا list أصلاً
    (لا DataFrame)، فـ_summarize_report في سجلّ التجارب تستخدمها كما هي."""
    pass


register_hypothesis(
    hyp_id="H001_short_term_reversal",
    hypothesis=(
        "انعكاس قصير المدى (أفق يوم واحد): آخر تغيّر في السعر يتنبأ بعكس "
        "اتجاهه في العائد التالي — لا استمراره (الزخم). اكتُشفت أثناء "
        "اختبار سلامة محور التقييم بفرضية الزخم البسيط؛ الإشارة خرجت "
        "معكوسة باتساق إحصائي قوي بدل الاتجاه المتوقَّع."
    ),
    source="literature_mining",   # مُدرَجة أصلاً في جدول فرضيات الأساس (مسار ج)
    status="مقبولة",
    report=_actual_report,
    notes=(
        "22/30 نافذة سالبة، IC متوسط=-0.0805 (انحراف=0.0835)، نوافذ معنوية "
        "بنفس اتجاه المتوسط=73%. consistent_sign=False لأن نافذتين (14، 18) "
        "خرجتا موجبتين بدلالة قوية (p=0.000، percentile=100) — استثناء "
        "حقيقي لا ضجيج، قيد الفحص كمرشّح لفرضية 'تغيّر النظام' (قسم ١١). "
        "monotonic=False في كل نافذة متوقَّع (معيار أشدّ صرامة من IC نفسه)، "
        "لا ينفي القوة."
    ),
)

list_registry()


✅ سُجّلت 'H001_short_term_reversal' بحالة 'مقبولة' في experiment_registry/registry.json


,id,hypothesis,source,status,mean_ic,frac_significant,consistent_sign
0,H001_short_term_reversal,انعكاس قصير المدى (أفق يوم واحد): آخر تغيّر في...,literature_mining,مقبولة,-0.0805,0.73,False


## ١١) فحص نافذة بعينها — تشخيص أنظمة السوق (النافذتان الشاذّتان ١٤، ١٨)

أداتان عامّتان (تُستخدَمان لأي نافذة لاحقاً، لا لهاتين فقط): `inspect_window`
تستخرج تواريخ ونِسَب `train`/`val`/`test` الفعلية من `windows[n-1]` مباشرة —
بلا حاجة لحفظ `schedule` بمعزل. `regime_snapshot` تُعطي لقطة خام (اتجاه عام
+ تقلّب "آخر تغيّر") لعيّنات `test` في تلك النافذة — تشخيص أوّلي فقط، لا
بديل عن نظام أنظمة سوق كامل (HMM/GMM، المرحلة ٢ في خطة المشروع).

تتطلّبان `sample_timestamps`/`embargo_duration` (من دفتر التحضير) و
`concat_splits`/`extract_feature_last_diff` (قسم ٦ أعلاه) متاحة في نفس
الجلسة — كبقية أمثلة هذا القسم.

In [ ]:
# @title
def inspect_window(windows, n: int, window_names=None) -> dict:
    """تواريخ ونِسَب train/val/test الفعلية للنافذة رقم n (١-محسوبة) من
    windows الناتجة عن rolling_splits — نفس القائمة المُمرَّرة أصلاً لـ
    evaluate_hypothesis_over_rolling_windows، بلا حاجة لحفظ schedule بمعزل."""
    if not (1 <= n <= len(windows)):
        raise IndexError(f"النافذة {n} خارج المدى (المتاح 1..{len(windows)}).")
    train, val, test = windows[n - 1]
    name = window_names[n - 1] if window_names else f"نافذة {n}"

    tr_ts = sample_timestamps(concat_splits(train))
    va_ts = sample_timestamps(concat_splits(val))
    te_ts = sample_timestamps(concat_splits(test))

    tr_start, tr_end, tr_n = tr_ts.min(), tr_ts.max(), len(tr_ts)
    va_start, va_end, va_n = va_ts.min(), va_ts.max(), len(va_ts)
    te_start, te_end, te_n = te_ts.min(), te_ts.max(), len(te_ts)

    info = {"name": name,
            "train_range": (tr_start, tr_end, tr_n),
            "val_range": (va_start, va_end, va_n),
            "test_range": (te_start, te_end, te_n)}
    print(f"🔁 {name}: "
          f"train {tr_start:%Y-%m-%d}→{tr_end:%Y-%m-%d} ({tr_n:,}) | "
          f"val {va_start:%Y-%m-%d}→{va_end:%Y-%m-%d} ({va_n:,}) | "
          f"test {te_start:%Y-%m-%d}→{te_end:%Y-%m-%d} ({te_n:,})")
    return info


def regime_snapshot(windows, n: int, feature: str = "close", tf=None,
                    feature_order=None) -> dict:
    """لقطة نظام سوقي خام لعيّنات test في النافذة n: اتجاه عام (متوسط 'آخر
    تغيّر' في السعر) وتقلّبه (انحراف معياري) — يفحص هل نافذة شاذّة (IC معكوس
    عن بقية النوافذ) تقع في سوق متّجه بقوة أو متذبذب. مرّر feature_order من
    dataset['feature_order'] الأصلي (غير محفوظة داخل النافذة نفسها)."""
    _, _, test = windows[n - 1]
    diffs = extract_feature_last_diff(test, feature=feature, tf=tf, feature_order=feature_order)
    mean_diff = float(np.mean(diffs))
    std_diff = float(np.std(diffs))
    frac_pos = float(np.mean(diffs > 0))
    snap = {"n_samples": len(diffs), "mean_last_diff": mean_diff,
           "std_last_diff": std_diff, "frac_positive": frac_pos}
    print(f"   اتجاه عام={mean_diff:+.4f} | تقلّب={std_diff:.4f} | "
          f"نسبة عيّنات صاعدة={frac_pos:.0%}")
    return snap


# شغّل هذا بعد بناء windows (قسم ٨) في نفس الجلسة:
for n in (14, 18):
    inspect_window(windows, n)
    regime_snapshot(windows, n, feature_order=dataset["feature_order"])


🔁 نافذة 14: train 2019-11-13→2021-06-24 (18,564) | val 2021-07-28→2021-08-11 (1,080) | test 2021-09-14→2021-10-13 (2,220)
   اتجاه عام=+0.0070 | تقلّب=0.5064 | نسبة عيّنات صاعدة=49%
🔁 نافذة 18: train 2019-11-13→2021-10-22 (27,323) | val 2021-11-25→2021-12-09 (1,218) | test 2022-01-12→2022-02-10 (2,557)
   اتجاه عام=-0.0372 | تقلّب=0.3195 | نسبة عيّنات صاعدة=52%


### تواريخ النافذتين ١٤ و١٨ — مُشتقّة رياضياً من جدولك الفعلي، لا مُخمَّنة

الدفتر لا يملك بياناتك الحقيقية (Drive/Colab)، فتعذّر تشغيل `inspect_window`
هنا فعلياً. لكن جدول `rolling_split_schedule` **حتمي**: كل نافذة تتقدّم عن
سابقتها بمقدار `step` ثابت، وإزاحتا `val_end`/`test_end` عن `train_end`
ثابتتان عبر كل النوافذ — وكلها معروفة **مباشرة من ثلاث نوافذ حقيقية أرسلتَها**
(النوافذ ٢٨، ٢٩، ٣٠). استُخرجت `step=30` يوماً، وإزاحة `val_end`=+48 يوماً،
وإزاحة `test_end`=+63 يوماً — **بلا أي افتراض عن CONFIG** (الإزاحتان
مُقاستان من فروق تواريخك الفعلية، لا محسوبتين من `window_size`/`embargo`
افتراضياً) — ثم طُبِّقت الصيغة عكسياً لإيجاد `train_end` النافذة الأولى،
وتحقّقت بإعادة إنتاج تواريخ النوافذ الثلاث الثلاثة **حرفياً**:

| النافذة | train≤ | val≤ | test≤ |
|---|---|---|---|
| ١٤ | 2021-06-24 | 2021-08-11 | **2021-10-13** |
| ١٨ | 2021-10-22 | 2021-12-09 | **2022-02-10** |

**قراءة أوّلية بالسياق العام المعروف لسوق الكريبتو في هاتين الفترتين** (معرفة
عامة، لا قياساً على بياناتك — `regime_snapshot` أعلاه هو التحقّق الفعلي):

* **نافذة ١٤ (تنتهي 2021-10-13):** تسبق مباشرة موجة صعود أكتوبر 2021 القوية
  (إطلاق أوّل ETF لعقود BTC الآجلة في 19 أكتوبر، وقمة تاريخية جديدة ~٦٧
  ألف$ في 20 أكتوبر) — عقب تصحيح حاد أواخر سبتمبر (تشديد الصين حظر
  الكريبتو، ٢٤ سبتمبر). فترة **اتجاه صاعد قوي**، لا تذبذب مسطّح.
* **نافذة ١٨ (تنتهي 2022-02-10):** تقع مباشرة داخل هبوط يناير 2022 الحاد —
  BTC من ~٤٧ ألف$ (أوّل يناير) إلى أقل من ٣٣ ألف$ (٢٤ يناير)، مدفوعاً بمحضر
  اجتماع الفيدرالي (٥ يناير) وإشارات تشديد نقدي أسرع. فترة **اتجاه هابط حاد**.

كلا الفترتين اتجاه قوي مستمر، لا تذبذب — يطابق تماماً التفسير الأرجح: فرضية
الانعكاس تضعف أو تنعكس هي نفسها في أنظمة سوق **متّجهة بقوة** (حيث الاستمرار
يهزم الارتداد)، بعكس بقية النوافذ الأهدأ نسبياً. هذا **تفسير مبدئي من
التاريخ العام المعروف، لا قياساً رقمياً على تقلّب/اتجاه بياناتك الفعلية** —
شغّل `regime_snapshot(windows, 14, ...)` و`regime_snapshot(windows, 18, ...)`
في جلستك لتأكيده رقمياً، وهذا بحدّ ذاته أول دليل تجريبي مباشر لفرضية "تغيّر
النظام كميزة سياقية" (جدول الفرضيات الاستكشافية، خطة المشروع) — إن تأكّد
الرقم، فهذه فرضية ثانية جاهزة للتسجيل في سجلّ التجارب، لا مجرّد ملاحظة جانبية.

## ١٢) مقارنة بخط أساس، وتحديث ملاحظات H001 بالنتيجة الفعلية

نتيجة `regime_snapshot` الفعلية **لا تدعم بوضوح** فرضية "اتجاه سوقي قوي"
كما صيغت أوّلاً — وهذا يستحق تصحيحاً صريحاً، لا تجاهلاً:

| النافذة | اتجاه عام | تقلّب | % صاعدة | IC |
|---|---|---|---|---|
| ١٤ | +0.0070 (شبه صفر) | 0.5064 (الأعلى) | 49% (متوازن) | +0.080 |
| ١٨ | -0.0372 (هابط) | 0.3195 | 52% | +0.087 |

* **نافذة ١٨ تتّفق مع تاريخها المعروف**: اتجاه سالب يطابق هبوط يناير 2022،
  و52% عيّنات صاعدة رغم متوسط سالب يشير لتوزيع منحرف — كثير مكاسب صغيرة
  تقابلها خسائر أكبر حجماً وأقل تكراراً (نمط منسجم مع هبوط تدريجي تتخلّله
  أيام انهيار حاد، لا انزلاق مستقيم).
* **نافذة ١٤ تناقض الفرضية الأصلية**: لم تكن بعد داخل اتجاه صاعد قوي عند
  إغلاقها (13 أكتوبر) — اتجاهها العام شبه صفري وتوازن صاعد/هابط شبه تام؛
  الصعود القوي (ETF، قمة تاريخية) بدأ **بعد** انتهاء هذه النافذة (19-20
  أكتوبر)، لا خلالها. ما يميّزها فعلياً هو **أعلى تقلّب** بين النافذتين —
  سوق مضطرب غير محسوم الاتجاه، لا سوق متّجه بثبات.

**تصحيح الفرضية:** القاسم المشترك الأرجح ليس "اتجاه قوي" بل **قرب حدث سوقي
حاسم** (تشديد الصين للحظر أواخر سبتمبر → نافذة ١٤؛ محضر الفيدرالي المتشدّد
5 يناير → نافذة ١٨) — فترات اضطراب/تحوّل نظام، لا اتجاه هادئ مستمر بالضرورة.
هذا لا يزال يحتاج **خط أساس للمقارنة**: هل نوافذ الانعكاس العادية (IC سالب
بقوة) تختلف فعلاً في هذه الإحصاءات عن ١٤/١٨، أم أن التقلّب هنا طبيعي وليس
استثنائياً؟ بلا هذه المقارنة، الاستنتاج أعلاه وصف لا دليل حاسم.

In [ ]:
# @title
# خط أساس للمقارنة: أقوى ثلاث نوافذ انعكاس (IC الأكثر سلبية) — النافذتان
# 21 و26 و10 (IC = -0.189, -0.215, -0.180 على التوالي).
for n in (10, 21, 26):
    inspect_window(windows, n)
    regime_snapshot(windows, n, feature_order=dataset["feature_order"])
    print()

# بعد رؤية الأرقام: حدّث سجلّ H001 بالنتيجة الفعلية (لا الفرضية الأصلية غير
# المؤكَّدة) — يستبدل الملاحظات بالكامل بنفس المعرّف، فلا يبقى نصّ قديم مضلّل.
register_hypothesis(
    hyp_id="H001_short_term_reversal",
    hypothesis=(
        "انعكاس قصير المدى (أفق يوم واحد): آخر تغيّر في السعر يتنبأ بعكس "
        "اتجاهه في العائد التالي — لا استمراره (الزخم). اكتُشفت أثناء "
        "اختبار سلامة محور التقييم بفرضية الزخم البسيط؛ الإشارة خرجت "
        "معكوسة باتساق إحصائي قوي بدل الاتجاه المتوقَّع."
    ),
    source="literature_mining",
    status="مقبولة",
    report=_actual_report,
    notes=(
        "22/30 نافذة سالبة، IC متوسط=-0.0805 (انحراف=0.0835)، نوافذ معنوية "
        "بنفس اتجاه المتوسط=73%. نافذتان (14، 18) موجبتان بدلالة قوية "
        "(p=0.000، percentile=100). regime_snapshot الفعلي عليهما: "
        "نافذة 14 (test ينتهي 2021-10-13) اتجاه شبه صفري (+0.007) وأعلى "
        "تقلّب (0.51) وتوازن صاعد/هابط (49%) — سوق مضطرب غير محسوم، لا "
        "اتجاه قوي كما افتُرض أوّلاً؛ الصعود التاريخي (ETF/قمة أكتوبر 2021) "
        "بدأ بعد إغلاق النافذة لا خلالها. نافذة 18 (test ينتهي 2022-02-10) "
        "اتجاه سالب (-0.037) وتقلّب أقلّ (0.32) — يتّفق مع هبوط يناير 2022 "
        "المعروف. الفرضية المصحَّحة: قرب حدث سوقي حاسم (تشديد الصين، محضر "
        "الفيدرالي) لا 'اتجاه قوي' بذاته. غير مؤكَّد بعد — ينقصه خط أساس من "
        "نوافذ انعكاس عادية (راجع الخلية أعلاه) قبل تسجيل فرضية 'تغيّر "
        "النظام' كسجلّ مستقل."
    ),
)


🔁 نافذة 10: train 2019-11-13→2021-02-24 (11,156) | val 2021-03-30→2021-04-13 (884) | test 2021-05-17→2021-06-15 (2,050)
   اتجاه عام=-0.0617 | تقلّب=0.4948 | نسبة عيّنات صاعدة=45%

🔁 نافذة 21: train 2019-11-13→2022-01-20 (34,596) | val 2022-02-23→2022-03-09 (1,306) | test 2022-04-12→2022-05-11 (2,738)
   اتجاه عام=-0.0914 | تقلّب=0.4341 | نسبة عيّنات صاعدة=41%

🔁 نافذة 26: train 2019-11-13→2022-06-19 (48,061) | val 2022-07-23→2022-08-06 (1,442) | test 2022-09-09→2022-10-08 (2,910)
   اتجاه عام=-0.0337 | تقلّب=0.4617 | نسبة عيّنات صاعدة=45%

✅ سُجّلت 'H001_short_term_reversal' بحالة 'مقبولة' في experiment_registry/registry.json


{'id': 'H001_short_term_reversal',
 'hypothesis': 'انعكاس قصير المدى (أفق يوم واحد): آخر تغيّر في السعر يتنبأ بعكس اتجاهه في العائد التالي — لا استمراره (الزخم). اكتُشفت أثناء اختبار سلامة محور التقييم بفرضية الزخم البسيط؛ الإشارة خرجت معكوسة باتساق إحصائي قوي بدل الاتجاه المتوقَّع.',
 'source': 'literature_mining',
 'status': 'مقبولة',
 'notes': "22/30 نافذة سالبة، IC متوسط=-0.0805 (انحراف=0.0835)، نوافذ معنوية بنفس اتجاه المتوسط=73%. نافذتان (14، 18) موجبتان بدلالة قوية (p=0.000، percentile=100). regime_snapshot الفعلي عليهما: نافذة 14 (test ينتهي 2021-10-13) اتجاه شبه صفري (+0.007) وأعلى تقلّب (0.51) وتوازن صاعد/هابط (49%) — سوق مضطرب غير محسوم، لا اتجاه قوي كما افتُرض أوّلاً؛ الصعود التاريخي (ETF/قمة أكتوبر 2021) بدأ بعد إغلاق النافذة لا خلالها. نافذة 18 (test ينتهي 2022-02-10) اتجاه سالب (-0.037) وتقلّب أقلّ (0.32) — يتّفق مع هبوط يناير 2022 المعروف. الفرضية المصحَّحة: قرب حدث سوقي حاسم (تشديد الصين، محضر الفيدرالي) لا 'اتجاه قوي' بذاته. غير مؤكَّد بعد — ينقصه خط أساس من نوافذ انع

## ١٣) تصحيح: فرضية "قرب حدث سوقي حاسم" لا تصمد أمام خط الأساس

مقارنة فعلية بالأرقام (نوافذ ١٠/٢١/٢٦ هي **أقوى ثلاث نوافذ انعكاس** — IC
الأكثر سلبية بين الثلاثين):

| النافذة | IC | اتجاه عام | تقلّب | % صاعدة |
|---|---|---|---|---|
| ١٤ (شاذّة) | +0.080 | +0.0070 | **0.5064** | 49% |
| ١٨ (شاذّة) | +0.087 | -0.0372 | 0.3195 | 52% |
| ١٠ (أساس) | -0.180 | -0.0617 | 0.4948 | 45% |
| ٢١ (أساس) | -0.189 | -0.0914 | 0.4341 | 41% |
| ٢٦ (أساس) | -0.215 | -0.0337 | 0.4617 | 45% |

لا مقياس من الثلاثة يفصل ١٤/١٨ عن خط الأساس باتجاه ثابت:

* **التقلّب**: نوافذ الأساس كلها بين 0.43-0.49. نافذة ١٤ (0.51) بالكاد
  أعلاها — فرق هامشي لا يستحقّ وصفه "الأعلى بوضوح" كما قلتُ سابقاً. نافذة
  ١٨ (0.32) **أقلّ** من كل نوافذ الأساس — عكس اتجاه فرضية "تقلّب أعلى" تماماً.
* **الاتجاه العام**: نافذة ١٨ (-0.037) شبه مطابقة لنافذة الأساس ٢٦ (-0.034) —
  نفس الاتجاه والحجم تقريباً، لكن IC متعاكس تماماً (+0.087 مقابل -0.215).
  لو كان الاتجاه العام هو المُفسِّر، لاتّفقت هاتان النافذتان في IC، ولم
  يحدث.
* **% صاعدة**: نافذة ١٨ (52%) فعلاً أعلى من كل نوافذ الأساس (41-45%) — أقرب
  إشارة تمييز حقيقية وُجدت، لكنها وحيدة (لا تدعمها نافذة ١٤: 49%، قريبة من
  المدى العادي) ومن بيانات قليلة جداً (نافذتان فقط) لبناء استنتاج عليها.

**الخلاصة الصادقة: التفسير الذي اقترحتُه ("قرب حدث سوقي حاسم") مفنَّد فعلياً
بهذه المقارنة — لا يفسّر البيانات.** الشذوذ (نافذتان من 30 بإشارة معاكسة
قوية) لا يزال **غير مُفسَّر** بإحصاءات `regime_snapshot` الخام (اتجاه/تقلّب/
نسبة صاعدة). هذا لا يُضعف H001 نفسها — 28/30 نافذة (93%) لا تزال متّسقة،
والدليل الإحصائي على الانعكاس قوي بذاته — لكنه يعني أن "لماذا هاتان
النافذتان تحديداً" سؤال مفتوح حقيقي، لا محلولاً كما بدا سابقاً.

**الخطوة الصحيحة التالية إن أردتَ ملاحقته فعلياً**: ليست مزيداً من لقطات
يدوية لنوافذ مُختارة يدوياً (تعرّض حقيقي لتحيّز اختيار العيّنة) — بل أداة
كشف الأنظمة غير المُشرَفة (HMM/GMM على التقلّب والاتجاه) المُدرَجة أصلاً في
خطة المشروع، قسم "أدوات ونماذج لاكتشاف نمط غير معروف"، مرحلة ٣. أرخص بديل
عملي إن أردتَ الاستمرار الآن دون بناء ذلك: تشغيل `regime_snapshot` على
الثلاثين نافذة كلها (لا خمس مُنتقاة) ثم فحص هل ١٤/١٨ فعلاً قيمتان متطرّفتان
إحصائياً على أي عمود (percentile خارج التوزيع الكامل)، لا مجرّد مقارنة بثلاث
نوافذ يدوية.

In [ ]:
# @title
# خيار أسرع من بناء HMM/GMM كاملة: شغّل regime_snapshot على الثلاثين نافذة
# كلها (لا خمس مُنتقاة يدوياً)، ثم افحص أين تقع 14/18 فعلياً ضمن التوزيع
# الكامل — تحيّز اختيار العيّنة هو أكبر ثغرة في المقارنة اليدوية أعلاه.
_all_snaps = []
for n in range(1, len(windows) + 1):
    s = regime_snapshot(windows, n, feature_order=dataset["feature_order"])
    s["n"] = n
    _all_snaps.append(s)

import pandas as pd
_regime_table = pd.DataFrame(_all_snaps)[["n", "n_samples", "mean_last_diff", "std_last_diff", "frac_positive"]]
_regime_table["std_percentile"] = _regime_table["std_last_diff"].rank(pct=True) * 100
_regime_table["mean_percentile"] = _regime_table["mean_last_diff"].rank(pct=True) * 100
print(_regime_table.to_string(index=False))
print()
print("نافذة 14 — مئينيّ التقلّب:", _regime_table.loc[_regime_table["n"] == 14, "std_percentile"].iloc[0])
print("نافذة 18 — مئينيّ التقلّب:", _regime_table.loc[_regime_table["n"] == 18, "std_percentile"].iloc[0])


   اتجاه عام=-0.0389 | تقلّب=0.5765 | نسبة عيّنات صاعدة=52%
   اتجاه عام=+0.0038 | تقلّب=0.3470 | نسبة عيّنات صاعدة=52%
   اتجاه عام=+0.0391 | تقلّب=0.5517 | نسبة عيّنات صاعدة=50%
   اتجاه عام=+0.0769 | تقلّب=0.7101 | نسبة عيّنات صاعدة=54%
   اتجاه عام=+0.1310 | تقلّب=0.8092 | نسبة عيّنات صاعدة=54%
   اتجاه عام=+0.2002 | تقلّب=0.7869 | نسبة عيّنات صاعدة=58%
   اتجاه عام=+0.0523 | تقلّب=0.3921 | نسبة عيّنات صاعدة=53%
   اتجاه عام=+0.1264 | تقلّب=0.6347 | نسبة عيّنات صاعدة=55%
   اتجاه عام=-0.0115 | تقلّب=0.6062 | نسبة عيّنات صاعدة=47%
   اتجاه عام=-0.0617 | تقلّب=0.4948 | نسبة عيّنات صاعدة=45%
   اتجاه عام=-0.0544 | تقلّب=0.3994 | نسبة عيّنات صاعدة=49%
   اتجاه عام=+0.1180 | تقلّب=0.4366 | نسبة عيّنات صاعدة=63%
   اتجاه عام=+0.0142 | تقلّب=0.5937 | نسبة عيّنات صاعدة=54%
   اتجاه عام=+0.0070 | تقلّب=0.5064 | نسبة عيّنات صاعدة=49%
   اتجاه عام=+0.0824 | تقلّب=0.8299 | نسبة عيّنات صاعدة=53%
   اتجاه عام=-0.0777 | تقلّب=0.5468 | نسبة عيّنات صاعدة=46%
   اتجاه عام=-0.0386 | تقلّب=0.3543 | نس

### تحديث ملاحظات H001 بالخلاصة الصادقة (لا الفرضية المفنَّدة)

In [ ]:
# @title
register_hypothesis(
    hyp_id="H001_short_term_reversal",
    hypothesis=(
        "انعكاس قصير المدى (أفق يوم واحد): آخر تغيّر في السعر يتنبأ بعكس "
        "اتجاهه في العائد التالي — لا استمراره (الزخم). اكتُشفت أثناء "
        "اختبار سلامة محور التقييم بفرضية الزخم البسيط؛ الإشارة خرجت "
        "معكوسة باتساق إحصائي قوي بدل الاتجاه المتوقَّع."
    ),
    source="literature_mining",
    status="مقبولة",
    report=_actual_report,
    notes=(
        "28/30 نافذة (93%) متّسقة سالبة، IC متوسط=-0.0805 (انحراف=0.0835)، "
        "نوافذ معنوية بنفس اتجاه المتوسط=73%. نافذتان (14، 18) موجبتان "
        "بدلالة قوية (p=0.000، percentile=100) — شذوذ حقيقي، غير مُفسَّر. "
        "فُحصت فرضيتان تفسيريتان وفُنِّدتا بالبيانات: (أ) 'اتجاه سوقي قوي' — "
        "نافذة 14 كانت شبه مسطّحة فعلياً (mean=+0.007) لا صاعدة. (ب) 'قرب "
        "حدث سوقي حاسم/تقلّب أعلى' — نافذة 18 تقلّبها (0.32) أقلّ من كل "
        "نوافذ الأساس المقارَنة (0.43-0.49)، ونافذة 18 ومطابقتها بالاتجاه "
        "العام لنافذة أساس (26، -0.034 مقابل -0.037) رغم IC متعاكس تماماً. "
        "الشذوذ يبقى غير مُفسَّر بإحصاءات regime_snapshot الخام؛ الخطوة "
        "التالية الصحيحة كشف أنظمة غير مُشرَف (HMM/GMM، مرحلة 3 في الخطة) "
        "لا مزيد من مقارنات يدوية لنوافذ مُنتقاة (تحيّز اختيار عيّنة). "
        "لا يُضعف هذا الحكم على H001 نفسها — 93% اتساق دليل قوي بذاته."
    ),
)


✅ سُجّلت 'H001_short_term_reversal' بحالة 'مقبولة' في experiment_registry/registry.json


{'id': 'H001_short_term_reversal',
 'hypothesis': 'انعكاس قصير المدى (أفق يوم واحد): آخر تغيّر في السعر يتنبأ بعكس اتجاهه في العائد التالي — لا استمراره (الزخم). اكتُشفت أثناء اختبار سلامة محور التقييم بفرضية الزخم البسيط؛ الإشارة خرجت معكوسة باتساق إحصائي قوي بدل الاتجاه المتوقَّع.',
 'source': 'literature_mining',
 'status': 'مقبولة',
 'notes': "28/30 نافذة (93%) متّسقة سالبة، IC متوسط=-0.0805 (انحراف=0.0835)، نوافذ معنوية بنفس اتجاه المتوسط=73%. نافذتان (14، 18) موجبتان بدلالة قوية (p=0.000، percentile=100) — شذوذ حقيقي، غير مُفسَّر. فُحصت فرضيتان تفسيريتان وفُنِّدتا بالبيانات: (أ) 'اتجاه سوقي قوي' — نافذة 14 كانت شبه مسطّحة فعلياً (mean=+0.007) لا صاعدة. (ب) 'قرب حدث سوقي حاسم/تقلّب أعلى' — نافذة 18 تقلّبها (0.32) أقلّ من كل نوافذ الأساس المقارَنة (0.43-0.49)، ونافذة 18 ومطابقتها بالاتجاه العام لنافذة أساس (26، -0.034 مقابل -0.037) رغم IC متعاكس تماماً. الشذوذ يبقى غير مُفسَّر بإحصاءات regime_snapshot الخام؛ الخطوة التالية الصحيحة كشف أنظمة غير مُشرَف (HMM/GMM، مرحلة 3 في الخطة) لا

## ١٤) الحكم النهائي: لا علاقة بين مقاييس النظام الخام وIC

اختبار كامل على الثلاثين نافذة (لا خمس مُنتقاة يدوياً — يزيل تحيّز اختيار
العيّنة نهائياً): ارتباط سبيرمان/بيرسون بين IC كل نافذة وكلٍّ من
`mean_last_diff`/`std_last_diff`/`frac_positive`.

| المقياس | سبيرمان r | p | بيرسون r | p |
|---|---|---|---|---|
| mean_last_diff | +0.115 | 0.543 | +0.119 | 0.530 |
| std_last_diff | -0.151 | 0.427 | -0.136 | 0.473 |
| frac_positive | +0.111 | 0.561 | +0.158 | 0.403 |

كل الارتباطات ضعيفة وغير معنوية إطلاقاً (p>0.4 للجميع). مقارنة مجموعتي
IC موجب/سالب مباشرة: متوسط التقلّب عند IC موجب=0.43 مقابل IC سالب=0.54 —
**أقلّ لا أعلى**، عكس أي رواية "تقلّب أعلى". **حكم حاسم: لا علاقة بين هذه
المقاييس الخام والإشارة.**

### تصحيح عدّ من الملاحظة الأولى في هذه المحادثة

"22 من 30 نافذة سالبة" (أول رسالة) كان خطأً — خُلط مع عدد النوافذ
**السالبة والمعنوية معاً**. العدّ الدقيق من `per_window` الكامل:

| | العدد |
|---|---|
| إشارة سالبة (أي درجة) | 25/30 |
| إشارة موجبة (أي درجة) | 5/30 — نوافذ ٢، ١٤، ١٧، ١٨، ٢٧ |
| سالبة **ومعنوية** (p<0.05) | 22/30 |
| موجبة **ومعنوية** (p<0.05) | **٢/30 فقط — نافذتا ١٤، ١٨** |

النوافذ الموجبة الثلاث الأخرى (٢، ١٧، ٢٧) غير معنوية (p=0.43، 0.38، 0.16) —
ضجيج محض، لا شذوذ حقيقي مثل ١٤/١٨.

In [30]:
# @title
from scipy import stats

_ic_by_n = {int(r["window"].split()[-1]): r["ic"] for r in _per_window_actual}
_ic_series = [_ic_by_n[n] for n in _regime_table["n"]]

for col in ("mean_last_diff", "std_last_diff", "frac_positive"):
    r_s, p_s = stats.spearmanr(_ic_series, _regime_table[col])
    r_p, p_p = stats.pearsonr(_ic_series, _regime_table[col])
    print(f"{col:16s} spearman r={r_s:+.3f} (p={p_s:.3f})   pearson r={r_p:+.3f} (p={p_p:.3f})")

import numpy as np
_ic_arr = np.array(_ic_series)
_pos = _ic_arr > 0
print()
print("متوسط التقلّب | IC>0:", _regime_table.loc[_pos, "std_last_diff"].mean(),
     " | IC<0:", _regime_table.loc[~_pos, "std_last_diff"].mean())


mean_last_diff   spearman r=+0.115 (p=0.543)   pearson r=+0.119 (p=0.530)
std_last_diff    spearman r=-0.151 (p=0.427)   pearson r=-0.136 (p=0.473)
frac_positive    spearman r=+0.111 (p=0.561)   pearson r=+0.158 (p=0.403)

متوسط التقلّب | IC>0: 0.43038941025733946  | IC<0: 0.5434071671962738


### تحديث أخير لسجلّ H001 — الصورة الكاملة الدقيقة، وتوصية بإيقاف هذا الفرع

In [31]:
# @title
register_hypothesis(
    hyp_id="H001_short_term_reversal",
    hypothesis=(
        "انعكاس قصير المدى (أفق يوم واحد): آخر تغيّر في السعر يتنبأ بعكس "
        "اتجاهه في العائد التالي — لا استمراره (الزخم). اكتُشفت أثناء "
        "اختبار سلامة محور التقييم بفرضية الزخم البسيط؛ الإشارة خرجت "
        "معكوسة باتساق إحصائي قوي بدل الاتجاه المتوقَّع."
    ),
    source="literature_mining",
    status="مقبولة",
    report=_actual_report,
    notes=(
        "25/30 إشارة سالبة، 22/30 سالبة ومعنوية (p<0.05) — هذا هو 73% "
        "(frac_significant)، لا 'العدد السالب' كما وُصف خطأً في أول تسجيل. "
        "5/30 موجبة الإشارة؛ منها 2 فقط معنويتان (نافذتا 14، 18، p=0.000، "
        "percentile=100) والثلاث الباقية (2، 17، 27) ضجيج محض غير معنوي. "
        "فُحصت 3 تفسيرات لشذوذ 14/18 وفُنِّدت كلها بالبيانات الكاملة: (أ) "
        "اتجاه سوقي قوي (نافذة 14 كانت شبه مسطّحة فعلياً)، (ب) قرب حدث "
        "سوقي/تقلّب أعلى (نافذة 18 تقلّبها أقلّ من نوافذ أساس مقارَنة)، (ج) "
        "ارتباط عام بين mean/std/frac_positive وIC عبر الثلاثين نافذة كلها "
        "(سبيرمان/بيرسون، كل p>0.4 — لا علاقة إحصائية). الشذوذ حقيقي "
        "(دلالة إحصائية قوية جداً في اتجاهين) لكن غير مُفسَّر بأي مقياس نظام "
        "خام جُرِّب. التوصية: إيقاف هذا المسار الفرعي هنا — استُنفدت الأدوات "
        "الرخيصة المعقولة (مقارنة يدوية + ارتباط كامل)؛ تفسيره يحتاج كشف "
        "أنظمة غير مُشرَف فعلي (HMM/GMM، مرحلة 3 بالخطة) لا مزيداً من "
        "الإحصاءات الوصفية. لا يمسّ هذا الحكم على H001 نفسها: 22/30 نافذة "
        "معنوية بنفس الاتجاه دليل قوي وكافٍ للقبول بذاته."
    ),
)


✅ سُجّلت 'H001_short_term_reversal' بحالة 'مقبولة' في experiment_registry/registry.json


{'id': 'H001_short_term_reversal',
 'hypothesis': 'انعكاس قصير المدى (أفق يوم واحد): آخر تغيّر في السعر يتنبأ بعكس اتجاهه في العائد التالي — لا استمراره (الزخم). اكتُشفت أثناء اختبار سلامة محور التقييم بفرضية الزخم البسيط؛ الإشارة خرجت معكوسة باتساق إحصائي قوي بدل الاتجاه المتوقَّع.',
 'source': 'literature_mining',
 'status': 'مقبولة',
 'notes': "25/30 إشارة سالبة، 22/30 سالبة ومعنوية (p<0.05) — هذا هو 73% (frac_significant)، لا 'العدد السالب' كما وُصف خطأً في أول تسجيل. 5/30 موجبة الإشارة؛ منها 2 فقط معنويتان (نافذتا 14، 18، p=0.000، percentile=100) والثلاث الباقية (2، 17، 27) ضجيج محض غير معنوي. فُحصت 3 تفسيرات لشذوذ 14/18 وفُنِّدت كلها بالبيانات الكاملة: (أ) اتجاه سوقي قوي (نافذة 14 كانت شبه مسطّحة فعلياً)، (ب) قرب حدث سوقي/تقلّب أعلى (نافذة 18 تقلّبها أقلّ من نوافذ أساس مقارَنة)، (ج) ارتباط عام بين mean/std/frac_positive وIC عبر الثلاثين نافذة كلها (سبيرمان/بيرسون، كل p>0.4 — لا علاقة إحصائية). الشذوذ حقيقي (دلالة إحصائية قوية جداً في اتجاهين) لكن غير مُفسَّر بأي مقياس نظام خام جُ